velero-2-mixin/targets.libsonnet

```diff
@@ -100,5 +100,115 @@ local utils = commonlib.utils {
+        '${' + vars.datasources.prometheus.name + '}',
+        'rate(velero_backup_success_total{%(queriesSelector)s}[$__rate_interval]) / clamp_min(rate(velero_backup_attempt_total{%(queriesSelector)s}[$__rate_interval]),0.001)' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    backupSize:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'velero_backup_tarball_size_bytes{%(queriesSelector)s}' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    backupTime:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'sum(increase(velero_backup_duration_seconds_bucket{le!="+Inf"}[$__interval:])) by (le)' % vars
+      ),
+
+
+    restoreSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_restore_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    restoreFailure:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_restore_failed_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('failure'),
+
+    restoreSuccessRate:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'rate(velero_restore_success_total{%(queriesSelector)s}[$__rate_interval]) / clamp_min(rate(velero_restore_attempt_total{%(queriesSelector)s}[$__rate_interval]),0.001)' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    volumeSnapshotSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_volume_snapshot_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    volumeSnapshotFailure:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_volume_snapshot_failure_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('failure'),
+
+    volumeSnapshotSuccessRate:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'rate(velero_volume_snapshot_success_total{%(queriesSelector)s}[$__rate_interval]) / clamp_min(rate(velero_volume_snapshot_attempt_total{%(queriesSelector)s}[$__rate_interval]),0.001)' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    csiSnapshotSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_csi_snapshot_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
```

Comment:
Do you still need a cluster label too for all of these with either the `success` or `failure` legend?

Reply:
Previously i would have said no, but now i realized that grafana dropped an empty attribute that comes alongisde metrics that can be populated: `schedule` with this realization I have a few choices.
1. Include cluster - instance - schedule
2. instance - schedule
3. schedule
Since these are all single selections for now. I think choice number 3 would be best... but for now i went with choice number 1. Let me know what you think. I will try out a couple of the other options and try to see which looks best.

Reply:
I think this works even though it is a little more verbose